contracts/wXTMBridge.sol

```diff
@@ -1,30 +1,47 @@
 // SPDX-License-Identifier: UNLICENSED
 pragma solidity ^0.8.22;
 
-import {EIP3009} from "./extensions/EIP3009.sol";
-import {ERC20Burnable} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
 
-import {IwXTM} from "./interfaces/IwXTM.sol";
+import { IERC20 } from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import { IwXTM } from "./interfaces/IwXTM.sol";
 
-contract wXTMBridge is EIP3009 {
-    address private immutable wXTM;
+contract wXTMBridge is Ownable {
+    address private wXTM;
 
-    event TokensMinted(address indexed to, uint256 indexed amount, bytes32 indexed authorizationId);
-    event TokensBurned(address indexed from, uint256 indexed amount, bytes32 indexed authorizationId);
+    event TokensUnwrapped(
+        address indexed from,
+        string tariAddress,
+        uint256 indexed amount,
+        bytes32 indexed authorizationId
+    );
 
-    constructor(address _wXTM, string memory version) EIP3009("wXTMBridge", version) {
+    constructor(address _wXTM, address _delegate) Ownable(_delegate) {
         wXTM = _wXTM;
     }
 
-    function _mintTokens(address to, uint256 amount) internal override {
-        IwXTM(wXTM).mint(to, amount);
+    function bridgeToTari(string memory tari, uint256 value) external {
+        IERC20(wXTM).transferFrom(msg.sender, address(this), value);
 
-        emit TokensMinted(to, amount, keccak256(abi.encode(to, amount, block.timestamp)));
-    }
+        IwXTM(wXTM).burn(address(this), value);
 
-    function _burnTokens(address from, uint256 amount) internal override {
-        IwXTM(wXTM).burn(from, amount);
+        emit TokensUnwrapped(msg.sender, tari, value, keccak256(abi.encode(address(this), value, block.timestamp)));
+    }
 
-        emit TokensBurned(from, amount, keccak256(abi.encode(from, amount, block.timestamp)));
+    function bridgeToTariWithAuthorization(
+        string memory tari,
+        uint256 value,
+        uint256 validAfter,
+        uint256 validBefore,
+        bytes32 nonce,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
+        IwXTM(wXTM).receiveWithAuthorization(msg.sender, address(this), value, validAfter, validBefore, nonce, v, r, s);
+
+        IwXTM(wXTM).burn(address(this), value);
+
+        emit TokensUnwrapped(msg.sender, tari, value, keccak256(abi.encode(msg.sender, value, block.timestamp)));
```

Comment:
I fail to see the usefullness of including the last parametert `keccak256(abi.encode(msg.sender, value, block.timestamp))` in the event. What is the point of including it?